.gitlab/hpsf-gitlab-ci.yml

```diff
@@ -39,3 +39,30 @@ INTEL-DATA-CENTER-MAX-1100:
       - build/result_pvc1100.xml
     reports:
       junit: build/result_pvc1100.xml
+
+AMD-MI300A:
+  stage: test
+  tags: [uo-gpu, odyssey, amd-mi300]
+  image: rocm/dev-ubuntu-24.04:6.2.4-complete
+  script:
+    - sudo apt-get update && sudo apt-get install -y cmake
+    - export GTEST_FILTER=-hip.unified_memory_zero_memset
+    - cmake
+      -B build
+      -DCMAKE_BUILD_TYPE=Release
+      -DCMAKE_CXX_COMPILER=hipcc
+      -DKokkos_ENABLE_HIP=ON
+      -DKokkos_ARCH_AMD_GFX942_APU=ON
```

Comment:
Out of curiosity, does it get properly autodetected?
If so which would we rather test for?

Reply:
Yes, it gets auto-detected. Since the `GitLab` builds are for specific architectures, I slightly prefer being explicit.